lib/Driver/ToolChains.cpp

```diff
@@ -1047,6 +1047,12 @@ toolchains::GenericUnix::constructInvocation(const LinkJobAction &job,
     break;
   case LinkKind::DynamicLibrary:
     Arguments.push_back("-shared");
+    if (getTriple().getOS() == llvm::Triple::Linux) {
+      if (getTriple().getSubArch() == llvm::Triple::SubArchType::ARMSubArch_v7 ||
+          getTriple().getSubArch() == llvm::Triple::SubArchType::ARMSubArch_v6) {
+        Arguments.push_back("-Wl,-Bsymbolic");
+      }
+    }
```

Comment:
Good find.  I'll check the llvm sources again and make sure that that getArch can only be arm when those subarch's are set then remove the redundant check for it.